src/infrastructure/strapi/problem-repository.ts

```diff
@@ -0,0 +1,72 @@
+import { ProblemRepository } from '@/domain/repositories/problem-repository';
+import { Problem } from '@/domain/entities/problem';
+import { ResultAsync, err, ok } from 'neverthrow';
+import { normalizeError } from '@/lib/err-utils';
+import { WithJson, withJson } from '../infra-utils';
+import { StrapiError } from './error';
+import { strapiProblems } from './problem-response';
+import { errorResponse } from '../dto/error';
+
+const STRAPI_API_URL =
+  (process.env.STRAPI_PUBLIC_URL || 'http://localhost:1337') + '/api';
```

Comment:
nits: process.env の値は `env.d.ts` のようなファイルで以下のように型定義することができますね
```TypeScript
declare module 'process' {
    global {
        namespace NodeJS {
            interface ProcessEnv {
                NODE_ENV: 'development' | 'production';

                JUDGE_API_ENDPOINT: string;
                JUDGE_API_KEY: string;
            }
        }
    }
}
```
ここではコード内に開発環境の URL がハードコードされているので、環境ごとに .env を切り替えるように変更するといいかと思います！
```suggestion
const STRAPI_API_URL = `{process.env.STRAPI_PUBLIC_URL}/api`
```

Reply:
こちらは修正箇所が多くなるので、
#45 で作成した constants ファイルで修正することにします。